packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'framework.dart';
+
+/// Return true to cancel the notification bubbling.
+typedef bool NotificationListenerCallback<T extends Notification>(T notification);
+
+abstract class Notification {
+  void dispatch(BuildContext target) {
+    target.visitAncestorElements((Element element) {
+      if (element is StatelessComponentElement &&
+          element.widget is NotificationListener) {
+        final NotificationListener widget = element.widget;
+        if (widget._dispatch(this))
+          return false;
+      }
+      return true;
+    });
+  }
+}
+
+class NotificationListener<T extends Notification> extends StatelessComponent {
```

Comment:
Ah, I knew there was _something_ I'd forgotten. I need to figure out how to make scrollable material work. Either require that scrollables contain their own Material, or use this Notification mechanism to have scrollables notify Material when they scroll, and move the Material outside the Scrollables.